lib/modules/dosomething/dosomething_northstar/dosomething_northstar.module

```diff
@@ -125,33 +125,36 @@ function dosomething_northstar_update_user($user, $password = null) {
 }
 
 /**
- * Get user profile data from Northstar for the given Drupal ID.
+ * Get user profile data from Northstar for the given ID.
  *
- * @param int $drupal_id Drupal user id.
+ * @param int $id Drupal user id.
+ * @param string $type The index type of the ID
  * @return object
  */
-function dosomething_northstar_get_northstar_user($drupal_id) {
-  $northstar_user_info = cache_get('northstar_user_info_' . $drupal_id, 'cache_dosomething_northstar');
+function dosomething_northstar_get_northstar_user($id, $type = 'drupal_id') {
+  $northstar_user_info = NULL;
```

Comment:
You might not need this at all...

if `cache_get()` doesn't find what you ask for, it'll return `NULL`. Which means even if an email gets passed into the string id you generate, it's not going to find it, and thus return `NULL`.